test/sai_test/sai_vlan_test.py

```diff
@@ -24,58 +24,358 @@
 from ptf.testutils import *
 from ptf.thriftutils import *
 from sai_utils import *
+import pdb
 
-class Vlan_Domain_Forwarding_Test(T0TestBase):
+class VlanMemberListTest(T0TestBase):
     """
-    Verify the basic VLAN forwarding.
-    In L2, if segement with VLAN tag and sends to a VLAN port, 
-    segment should be forwarded inside a VLAN domain.
+    This test verifies the VLAN member list using SAI_VLAN_ATTR_MEMBER_LIST
     """
+    def setUp(self):
+        T0TestBase.setUp(self, is_reset_default_vlan=False)
+
+    def runTest(self):
+        print("VlanMemberListTest")
+        mbr_list = []
+        mbr_list.extend(self.vlan_configer.get_vlan_member(self.vlans[10].vlan_oid))
+        mbr_list.extend(self.vlan_configer.get_vlan_member(self.vlans[20].vlan_oid))
+        self.assertEqual(len(mbr_list), 16)
+
+        for i in range(0, 8):
+            self.assertEqual(self.vlans[10].vlan_mport_oids[i], mbr_list[i])
+        for i in range(8, 16):
+            self.assertEqual(self.vlans[20].vlan_mport_oids[i - 8], mbr_list[i]) 
+
+        # Adding vlan members and veryfing vlan member list
+        new_vlan_member = sai_thrift_create_vlan_member(
+            self.client,
+            vlan_id=self.vlans[10].vlan_oid,
+            bridge_port_id=self.bridge_port_list[17],
+            vlan_tagging_mode=SAI_VLAN_TAGGING_MODE_UNTAGGED)
+
+        mbr_list = []
+        mbr_list.extend(self.vlan_configer.get_vlan_member(self.vlans[10].vlan_oid))
+        mbr_list.extend(self.vlan_configer.get_vlan_member(self.vlans[20].vlan_oid))
+        self.assertEqual(len(mbr_list), 17)
+
+        # Adding vlan members and veryfing vlan member list
+        for i in range(0, 8):
+            self.assertEqual(self.vlans[10].vlan_mport_oids[i], mbr_list[i])
+        self.assertEqual(new_vlan_member, mbr_list[8])
+        for i in range(9, 17):
+            self.assertEqual(self.vlans[20].vlan_mport_oids[i - 9], mbr_list[i]) 
+
+        # Removing vlan members and veryfing vlan member list
+        sai_thrift_remove_vlan_member(self.client, new_vlan_member)
+
+        mbr_list = []
+        mbr_list.extend(self.vlan_configer.get_vlan_member(self.vlans[10].vlan_oid))
+        mbr_list.extend(self.vlan_configer.get_vlan_member(self.vlans[20].vlan_oid))
+        self.assertEqual(len(mbr_list), 16)
+
+        for i in range(0, 8):
+            self.assertEqual(self.vlans[10].vlan_mport_oids[i], mbr_list[i])
+        for i in range(8, 16):
+            self.assertEqual(self.vlans[20].vlan_mport_oids[i - 8], mbr_list[i])
+
+    def tearDown(self):
+        sai_thrift_flush_fdb_entries(
```

Comment:
this flush fdb method should not be used as vlan member recovery, you should re-run add vlan member method